yt/convenience.py

```diff
@@ -6,132 +6,114 @@
+    fn = os.path.expanduser(fn)
+
+    if any(wildcard in fn for wildcard in "[]?!*"):
+        from yt.data_objects.time_series import DatasetSeries
+
+        return DatasetSeries(fn, *args, **kwargs)
+
+    if not (os.path.exists(fn) or fn.startswith("http")):
+        data_dir = ytcfg.get("yt", "test_data_dir")
+        alt_fn = os.path.join(data_dir, fn)
+        if os.path.exists(alt_fn):
+            fn = alt_fn
         else:
-            mylog.error("None of the arguments provided to load() is a valid file")
-            mylog.error("Please check that you have used a correct path")
-            raise YTOutputNotIdentified(args, kwargs)
-    for n, c in types_to_check.items():
-        if n is None:
-            continue
-        if c._is_valid(*args, **kwargs):
-            candidates.append(n)
-
-    # convert to classes
-    candidates = [output_type_registry[c] for c in candidates]
+            msg = "No such file or directory: %s" % fn
+            if os.path.exists(data_dir):
+                msg += "\n(Also tried %s)" % alt_fn
+            raise OSError(msg)
+
+    candidates = []
+    for cls in output_type_registry.values():
+        if cls._is_valid(fn, *args, **kwargs):
+            candidates.append(cls)
+
     # Find only the lowest subclasses, i.e. most specialised front ends
     candidates = find_lowest_subclasses(candidates)
+
     if len(candidates) == 1:
-        return candidates[0](*args, **kwargs)
-    if len(candidates) == 0:
-        if (
-            ytcfg.get("yt", "enzo_db") != ""
-            and len(args) == 1
-            and isinstance(args[0], str)
-        ):
-            erdb = EnzoRunDatabase()
-            fn = erdb.find_uuid(args[0])
-            n = "EnzoDataset"
-            if n in output_type_registry and output_type_registry[n]._is_valid(fn):
-                return output_type_registry[n](fn)
-        mylog.error("Couldn't figure out output type for %s", args[0])
-        raise YTOutputNotIdentified(args, kwargs)
-
-    mylog.error("Multiple output type candidates for %s:", args[0])
-    for c in candidates:
-        mylog.error("    Possible: %s", c)
-    raise YTOutputNotIdentified(args, kwargs)
-
-
-def simulation(parameter_filename, simulation_type, find_outputs=False):
+        return candidates[0](fn, *args, **kwargs)
+
+    if len(candidates) > 1:
+        mylog.error("Multiple output type candidates for %s:", fn)
+        for c in candidates:
+            mylog.error("    Possible: %s", c)
+
+    raise YTOutputNotIdentified(fn, args, kwargs)
+
+
+def simulation(fn, simulation_type, find_outputs=False):
     """
-    Loads a simulation time series object of the specified
-    simulation type.
+    Load a simulation time series object of the specified simulation type.
+
+    Parameters
+    ----------
+    fn : str, os.Pathlike, or byte (types supported by os.path.expandusers)
+        Name of the data file or directory.
+
+    simulation_type : str
+        E.g. 'Enzo'
```

Comment:
is `Enzo` caps in the simulation type?

Reply:
according to this test, yes
https://github.com/yt-project/yt/blob/fdc9210a35e41b45c114f271798ea46c2179d2eb/doc/source/cookbook/time_series_profiles.py#L4